### PR TITLE
allows marathon healthcheck interval and max consecutive failure to b…

### PR DESCRIPTION
**Pull request by shamsimam**
…e configurable

### CODE REVIEW

File: waiter/docs/parameters.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ Additional (optional) parameters that can be set:
 |`X-Waiter-Env`|none|A JSON map from string keys to string values|The keys and values correspond to environment variable names and values which will get passed to the command.|Use as needed.|
 |`X-Waiter-Expired-Instance-Restart-Rate`|0.1|(0-1]|The max percentage of total expired instances to recycle at the same time (e.g. if the service has 10 expired instances and a `X-Waiter-Expired-Instance-Restart-Rate` of 0.1, Waiter will only restart 1 at a time).|The default is reasonable for most cases.|
 |`X-Waiter-Grace-Period-Secs`|30|1-3600 (1 hour)|The amount of time to wait after a service starts before considering a health check failure to be an error.|Set this value to startup time of your service. If your service starts up within 30 seconds, don't bother with changing this value.|
+|`X-Waiter-Health-Check-Interval-Secs`|10|1-60 (1 minute)|The amount of time to wait between health checks.|Don't change it unless you have a good reason to do so.|
+|`X-Waiter-Health-Check-Max-Consecutive-Failures`|10|1-15|Number of consecutive health check failures after which the unhealthy instance will be terminated.|Don't change it unless you have a good reason to do so.|
```
Comment (dposada) [inline waiter/docs/parameters.md]: Isn't the default here 5 and not 10?

File: waiter/docs/parameters.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ Additional (optional) parameters that can be set:
 |`X-Waiter-Env`|none|A JSON map from string keys to string values|The keys and values correspond to environment variable names and values which will get passed to the command.|Use as needed.|
 |`X-Waiter-Expired-Instance-Restart-Rate`|0.1|(0-1]|The max percentage of total expired instances to recycle at the same time (e.g. if the service has 10 expired instances and a `X-Waiter-Expired-Instance-Restart-Rate` of 0.1, Waiter will only restart 1 at a time).|The default is reasonable for most cases.|
 |`X-Waiter-Grace-Period-Secs`|30|1-3600 (1 hour)|The amount of time to wait after a service starts before considering a health check failure to be an error.|Set this value to startup time of your service. If your service starts up within 30 seconds, don't bother with changing this value.|
+|`X-Waiter-Health-Check-Interval-Secs`|10|1-60 (1 minute)|The amount of time to wait between health checks.|Don't change it unless you have a good reason to do so.|
```
Comment (dposada) [inline waiter/docs/parameters.md]: Do we really want to allow health checks once per second?

File: waiter/src/waiter/service_description.clj [hunk 3]
```diff
@@ -67,6 +67,8 @@
    (s/optional-key "metric-group") schema/valid-metric-group
    ; start-up related
    (s/optional-key "health-check-url") schema/non-empty-string
+   (s/optional-key "health-check-interval-secs") (s/both s/Int (s/pred #(<= 1 % 60) 'at-most-1-minute))
+   (s/optional-key "health-check-max-consecutive-failures") (s/both s/Int (s/pred #(<= 1 % 15) 'at-most-ten))
```
Comment (dposada) [inline waiter/src/waiter/service_description.clj]: `'at-most-15`

**Comment by shamsimam** [comment]
@sradack any feedback on this?

**Pull request merged**